File: Backend/src/EbecShop.Model/Product.cs
using EbecShop.Model.Abstract;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace EbecShop.Model
{
    public class Product : Entity
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public byte[] Image { get; set; }

        public IEnumerable<ProductType> Types { get; set; }
    }
}

File: Frontend/src/app/product/product.ts
import { ProductType } from "./productType";

export interface IProduct{
    id: number;
    name: string;
    description: string;
    imageUrl: string;
    types: ProductType[];    
}

export class Product implements IProduct {

    constructor(public name: string,
                public description: string,
                public imageUrl: string,
                public types: ProductType[] = [new ProductType(null)],
                public id: number = 0)
    {
    }
}